compiler/compile/OSRData.cpp

```diff
@@ -1490,7 +1490,9 @@ bool TR_OSRCompilationData::TR_ScratchBufferInfo::operator==(const TR_ScratchBuf
       scratchBufferOffset == info.scratchBufferOffset && symSize == info.symSize;
    }
 
-#if defined(TR_HOST_POWER) && defined(TR_TARGET_POWER) && defined(__IBMCPP__)
+#if (defined(TR_HOST_POWER) && defined(TR_TARGET_POWER)                       \
+           || defined(TR_HOST_S390) && defined(TR_TARGET_S390))               \
+        && defined(__IBMCPP__)
 __attribute__((__noinline__))
 //This tiny function when inlined by xlC 12 at -O3 breaks java -version
```

Comment:
I believe you observed this issue with xlC 2.1.1 hence this comment has not aged well. Can you improve the comment and mention it was originally seen on 1.2 on Power but reproduced on 2.1.1 on z/OS so that we have more context?

Reply:
Thanks, Filip @fjeremic.  I've updated that comment in https://github.com/eclipse/omr/pull/3085/commits/42f72e85c25fa8cfd9d5c282835b72b8c419be69